tests/release/pipelines/fbc_release.go

```diff
@@ -15,186 +15,272 @@ import (
-	var releasePr *v1beta1.PipelineRun
-	var snapshot *appservice.Snapshot
+	var issueId = "bz12345"
+
+	AfterEach(framework.ReportFailure(&devFw))
 
 	stageOptions := utils.Options{
 		ToolchainApiUrl: os.Getenv(constants.TOOLCHAIN_API_URL_ENV),
 		KeycloakUrl:     os.Getenv(constants.KEYLOAK_URL_ENV),
 		OfflineToken:    os.Getenv(constants.OFFLINE_TOKEN_ENV),
 	}
 
-	BeforeAll(func() {
-
-		devFw, err = framework.NewFrameworkWithTimeout(
-			devWorkspace,
-			time.Minute*60,
-			stageOptions,
-		)
-		Expect(err).NotTo(HaveOccurred())
-
-		managedFw, err = framework.NewFrameworkWithTimeout(
-			managedWorkspace,
-			time.Minute*60,
-			stageOptions,
-		)
-		Expect(err).NotTo(HaveOccurred())
-
-		_, err = devFw.AsKubeDeveloper.HasController.CreateApplication(fbcApplicationName, devFw.UserNamespace)
-		Expect(err).NotTo(HaveOccurred())
-
-		componentObj := appservice.ComponentSpec{
-			ComponentName: fbcComponentName,
-			Application:   fbcApplicationName,
-			Source: appservice.ComponentSource{
-				ComponentSourceUnion: appservice.ComponentSourceUnion{
-					GitSource: &appservice.GitSource{
-						URL: fbcSourceGitUrl,
-					},
-				},
-			},
-			TargetPort: targetPort,
-		}
-		component, err = devFw.AsKubeDeveloper.HasController.CreateComponent(componentObj, devFw.UserNamespace, "", "", fbcApplicationName, false, map[string]string{})
-		GinkgoWriter.Println("component : ", component.Name)
-		Expect(err).ShouldNot(HaveOccurred())
+	Describe("with FBC happy path", Label("fbcHappyPath"), func() {
+		var component *appservice.Component
+		BeforeAll(func() {
 
-		_, err = devFw.AsKubeDeveloper.ReleaseController.CreateReleasePlan(fbcReleasePlanName, devNamespace, fbcApplicationName, managedNamespace, "true")
-		Expect(err).NotTo(HaveOccurred())
-
-		data, err := json.Marshal(map[string]interface{}{
-			"fbc": map[string]interface{}{
-				"fromIndex":            constants.FromIndex,
-				"targetIndex":          constants.TargetIndex,
-				"binaryImage":          constants.BinaryImage,
-				"requestUpdateTimeout": "420",
-				"buildTimeoutSeconds":  "480",
-			},
+			devFw, err = framework.NewFrameworkWithTimeout(
+				devWorkspace,
+				time.Minute*60,
+				stageOptions,
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			managedFw, err = framework.NewFrameworkWithTimeout(
+				managedWorkspace,
+				time.Minute*60,
+				stageOptions,
+			)
+			Expect(err).NotTo(HaveOccurred())
+			managedNamespace = managedFw.UserNamespace
+
+			_, err = devFw.AsKubeDeveloper.HasController.CreateApplication(fbcApplicationName, devNamespace)
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = devFw.AsKubeDeveloper.ReleaseController.CreateReleasePlan(fbcReleasePlanName, devNamespace, fbcApplicationName, managedNamespace, "true")
+			Expect(err).NotTo(HaveOccurred())
+
+			CreateReleasePlanAdmission(fbcReleasePlanAdmissionName, *managedFw, devNamespace, managedNamespace, fbcApplicationName, fbcEnterpriseContractPolicyName, relSvcCatalogPathInRepo, "false", "")
```

Comment:
why do we check for an error when creating the ReleasePlan but not ReleasePlanAdmission?

Reply:
The error is checked inside the function.